File: backend/app/schemas/outcome.py
from __future__ import annotations

from pydantic import BaseModel
from typing import Any


class LearnedFieldsSchema(BaseModel):
    company_size: int | None = None
    pain: int | None = None
    budget: bool | None = None
    authority: bool | None = None
    timeline: str | None = None


class TraceTurnSchema(BaseModel):
    turn_index: int
    user_text: str | None = None
    agent_text: str | None = None
    extracted: dict[str, Any] = {}
    action: dict[str, Any] = {}
    score_before: float = 0.0
    score_after: float = 0.0
    reasons: list[str] = []
    stage_before: str = ""
    stage_after: str = ""
    extracted_source: str = "rule_based"
    wording_source: str = "template"


class ScoreBreakdownItem(BaseModel):
    field: str
    points: float
    reason: str


class OutcomeSchema(BaseModel):
    learned_fields: LearnedFieldsSchema
    opportunity_score: float
    opportunity_label: str
    recommended_next_action: str
    summary: str
    score_breakdown: list[ScoreBreakdownItem] = []
    score_explanation: str = ""
    decision_trace: list[TraceTurnSchema] = []


class OutcomeResponse(BaseModel):
    status: str
    outcome: OutcomeSchema | None = None

File: frontend/src/types/outcome.ts
export interface LearnedFields {
  company_size: string;
  pain: string;
  budget: string;
  authority: string;
  timeline: string;
}

export interface ScoreBreakdownItem {
  field: string;
  points: number;
  reason: string;
}

export interface Outcome {
  session_id: string;
  learned_fields: LearnedFields;
  opportunity_score: number;
  opportunity_label: "Weak" | "Medium" | "Strong";
  recommended_next_action: string;
  summary: string;
  score_breakdown: ScoreBreakdownItem[];
  score_explanation: string;
}

export interface OutcomeResponse {
  status: "running" | "completed";
  outcome: Outcome | null;
}

export interface RunResponse {
  session_id: string;
  status: string;
  agent_text: string;
  prospect_mode: "human" | "ai";
  connect_info: { ws_url: string };
}

export interface TurnResponse {
  status: string;
  agent_text: string | null;
  prospect_text: string | null;
  opportunity_score: number | null;
  ended: boolean;
  outcome: Outcome | null;
}
